types/mousetrap/index.d.ts

```diff
@@ -4,35 +4,38 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
-interface ExtendedKeyboardEvent extends KeyboardEvent {
-    returnValue: boolean; // IE returnValue
-}
+declare namespace Mousetrap {
+    interface ExtendedKeyboardEvent extends KeyboardEvent {
+        returnValue: boolean; // IE returnValue
+    }
 
-interface MousetrapStatic {
-    (el: Element): MousetrapInstance;
-    new (el?: Element): MousetrapInstance;
-    addKeycodes(keycodes: { [key: number]: string }): void;
-    stopCallback: (e: ExtendedKeyboardEvent, element: Element, combo: string) => boolean;
-    bind(keys: string|string[], callback: (e: ExtendedKeyboardEvent, combo: string) => any, action?: string): MousetrapInstance;
-    unbind(keys: string|string[], action?: string): MousetrapInstance;
-    trigger(keys: string, action?: string): MousetrapInstance;
-    reset(): MousetrapInstance;
-
-    /** https://craig.is/killing/mice#extensions.global */
-    bindGlobal(keyArray: string|string[], callback: (e: ExtendedKeyboardEvent, combo: string) => any, action?: string): void;
-}
+    interface MousetrapStatic {
+        (el: Element): MousetrapInstance;
+        new (el?: Element): MousetrapInstance;
+        addKeycodes(keycodes: { [key: number]: string }): void;
+        stopCallback: (e: ExtendedKeyboardEvent, element: Element, combo: string) => boolean;
+        bind(keys: string|string[], callback: (e: ExtendedKeyboardEvent, combo: string) => any, action?: string): MousetrapInstance;
+        unbind(keys: string|string[], action?: string): MousetrapInstance;
+        trigger(keys: string, action?: string): MousetrapInstance;
+        reset(): MousetrapInstance;
+
+        /** https://craig.is/killing/mice#extensions.global */
+        bindGlobal(keyArray: string|string[], callback: (e: ExtendedKeyboardEvent, combo: string) => any, action?: string): void;
+    }
 
-interface MousetrapInstance {
-    stopCallback: (e: ExtendedKeyboardEvent, element: Element, combo: string) => boolean;
-    bind(keys: string|string[], callback: (e: ExtendedKeyboardEvent, combo: string) => any, action?: string): this;
-    unbind(keys: string|string[], action?: string): this;
-    trigger(keys: string, action?: string): this;
-    handleKey(character: string, modifiers: string[], e: ExtendedKeyboardEvent): void;
-    reset(): this;
+    interface MousetrapInstance {
```

Comment:
Thanks for opening the PR @OliverJAsh 🙂
I think this will introduce a breaking change right? Previous consumers of `MousetrapInstance` will now need to reference it as `Mousetrap.MousetrapInstance`?

Reply:
Correct. I don't think there's any way to avoid that.

Reply:
Thanks for the clarification, this looks good to me. If you like, you can add your github username to the top of `index.d.ts` as a contributor.

Reply:
I got this error in Angular 10:
This expression is not constructable.
  Type '{ default: MousetrapStatic; addKeycodes(keycodes: { [key: number]: string; }): void; stopCallback: (e: ExtendedKeyboardEvent, element: Element, combo: string) => boolean; ... 4 more ...; bindGlobal(keyArray: string | string[], callback: (e: ExtendedKeyboardEvent, combo: string) => any, action?: string): void; }' has no construct signatures.

Reply:
I got this error in Angular 10
This expression is not constructable.
  Type '{ default: MousetrapStatic; addKeycodes(keycodes: { [key: number]: string; }): void; stopCallback: (e: ExtendedKeyboardEvent, element: Element, combo: string) => boolean; ... 4 more ...; bindGlobal(keyArray: string | string[], callback: (e: ExtendedKeyboardEvent, combo: string) => any, action?: string): void; }' has no construct signatures.

Reply:
Since these are breaking changes, should these be released as a major release version instead of minor one? This will prevent, the libraries from which are dependent on this one, from getting upgraded automatically.

Reply:
Yes you're right, this shouldn't have been introduced with a PATCH version number - I made a mistake approving this. In the interim I'll revert with another minor version bump then re introduce with a major: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/48169

Reply:
IIUC, the official advice is to only bump the major version to match the version of the library the types are for.  I believe using a patch version for this change was the right thing to do. See https://github.com/DefinitelyTyped/DefinitelyTyped/issues/25677#issuecomment-388215959.